iosApp/iosApp/Pages/Search/Results/SearchResultsContainer.swift

```diff
@@ -154,32 +175,31 @@ struct SearchResultView_Previews: PreviewProvider {
     static var previews: some View {
         SearchResultsView(
```

Comment:
On the initial tap to open up the search results, I'm not seeing any recent visits until after I search for and clear the search field.

Unrelated, but also visible in the screenshot, the search being active no longer dismisses the sheet.

![Simulator Screenshot - iPhone 15 Pro - 2024-10-16 at 14 51 30](https://github.com/user-attachments/assets/18511b52-b71a-4103-aa65-470660234342)

Reply:
I think it might be! Looking into it 😄

Reply:
So I tried a few things to resolve this, first by just making sure some of this work was done on another thread which resolves the UI hanging but it was still super slow. So I looked into methods to optimize the route association on the Swift side but didn't really come up with anything. In the end it turns out that this is much more efficient if you just do it in Kotlin and call it from Swift!

Reply:
Nice, calling it from kotlin seems like the best approach anyway! The recently viewed loading looks good now 🎉 

I am still seeing the sheet displayed over the search results though. In the initial PR I'd added [this check](https://github.com/mbta/mobile_app/pull/429/files#diff-aa4e2052b22ab2454329f9761c75169e801d78127b9849bd18ed3d1e31c447aaR156-R158) to hide the sheet when the search was focused, that seems to still be in place so I'm not sure what's going wrong here...

Reply:
Oh, lol, we stopped using the sheet items at all. This should do it.
```swift
@ViewBuilder var mapWithSheets: some View {
        let nav = nearbyVM.navigationStack.lastSafe()
        let sheetItemId: String? = nav.sheetItemIdentifiable()?.id
        mapSection
            .sheet(isPresented: .constant(!(searchObserver.isSearching && nav == .nearby)), content: { ... }
```